configs/webpack.common.config.js

```diff
@@ -1,36 +1,76 @@
-    alias: {
-      bcoin$: `${MODULES_DIR}/bcoin/lib/bcoin-browser`,
-      bcash$: `${MODULES_DIR}/bcash/lib/bcoin-browser`,
-      hsd$: `${MODULES_DIR}/hsd/lib/hsd-browser`,
-      bledger: `${MODULES_DIR}/bledger/lib/bledger-browser`,
-      bmultisig: `${MODULES_DIR}/bmultisig/lib/bmultisig-browser`,
-      react: `${MODULES_DIR}/react`,
-      'react-router': `${MODULES_DIR}/react-router`,
-      'react-router-dom': `${MODULES_DIR}/react-router-dom`,
-      'react-redux': `${MODULES_DIR}/react-redux`,
-      'react-loadable': `${MODULES_DIR}/react-loadable`,
-      '&local': path.resolve(bpanelPrefix, 'local_plugins'),
-      '@bpanel/bpanel-utils': `${MODULES_DIR}/@bpanel/bpanel-utils`,
-      '@bpanel/bpanel-ui': `${MODULES_DIR}/@bpanel/bpanel-ui`,
-      tinycolor: 'tinycolor2'
-    }
+let MY_SECRETS = {};
+
+module.exports = () => {
+  try {
+    MY_SECRETS = require(path.resolve(bpanelPrefix, 'secrets.json'));
+  } catch (e) {
+    logger.error(
+      `Couldn't find secrets.json in ${bpanelPrefix}. Make sure to run npm install to create boilerplate files`
+    );
   }
+  return {
+    resolve: {
+      symlinks: false,
+      extensions: ['-browser.js', '.js', '.json', '.jsx'],
+      // list of aliases are what packages plugins can list as peerDeps
+      // this helps simplify plugin packages and ensures that parent classes
+      // all point to the same instance, e.g bcoin.TX will be same for all plugins
+      alias: {
+        bcoin$: `${MODULES_DIR}/bcoin/lib/bcoin-browser`,
+        bcash$: `${MODULES_DIR}/bcash/lib/bcoin-browser`,
+        hsd$: `${MODULES_DIR}/hsd/lib/hsd-browser`,
+        bledger: `${MODULES_DIR}/bledger/lib/bledger-browser`,
+        bmultisig: `${MODULES_DIR}/bmultisig/lib/bmultisig-browser`,
+        react: `${MODULES_DIR}/react`,
+        'react-router': `${MODULES_DIR}/react-router`,
+        'react-dom': `${MODULES_DIR}/react-dom`,
+        'react-router-dom': `${MODULES_DIR}/react-router-dom`,
+        'react-redux': `${MODULES_DIR}/react-redux`,
+        'react-loadable': `${MODULES_DIR}/react-loadable`,
+        '&local': path.resolve(bpanelPrefix, 'local_plugins'),
+        '@bpanel/bpanel-utils': `${MODULES_DIR}/@bpanel/bpanel-utils`,
+        '@bpanel/bpanel-ui': `${MODULES_DIR}/@bpanel/bpanel-ui`,
+        tinycolor: 'tinycolor2'
+      }
+    },
+    plugins: [
+      new HtmlWebpackPlugin({
+        title: 'bPanel - A Blockchain Management System',
+        template: `${path.join(SRC_DIR, 'index.template.ejs')}`,
+        inject: 'body'
+      }),
+      new WebpackShellPlugin({
+        onBuildStart: {
+          scripts: [
+            `node ${path.resolve(SERVER_DIR, 'clear-plugins.js')}`,
+            `node ${path.resolve(
+              SERVER_DIR,
+              'build-plugins.js'
+            )} --prefix=${bpanelPrefix}`
+          ]
+        }
+      }),
+      new webpack.DefinePlugin({
+        MY_SECRETS: JSON.stringify(MY_SECRETS),
```

Comment:
Why `MY_SECRETS` and not just `SECRETS`?

Reply:
No particular reason I guess. It felt a little more descriptive in that is specific to the secrets in your particular instance of bPanel (e.g. "my" ipstack API key for using the peers widget). I can change it though.

Reply:
I think `SECRETS` sounds more professional

Reply:
Noted! Updated to `SECRETS`